### ISSUE TITLE
Cannot call function with union of types although function is callable with each type alone

### ISSUE THREAD

**Issue by bersbersbers** [microsoft/pyright#9809]
**Describe the bug**
I can call `QKeySequence` with a `str` (`fun1`) or with a `QKeySequence.StandardKey` (`fun2`), but not with a union of the two (`fun3`). I hope it's not one of these union issues again...

**Code or Screenshots**

```python
from PySide6.QtGui import QKeySequence

def fun1(keys: str) -> None:
    QKeySequence(keys)

def fun2(keys: QKeySequence.StandardKey) -> None:
    QKeySequence(keys)

def fun3(keys: str | QKeySequence.StandardKey) -> None:
    QKeySequence(keys)
```

```
pyright bug.py
c:\Git\project\bug.py
  c:\Git\project\bug.py:10:5 - error: No overloads for "__init__" match the provided arguments (reportCallIssue)
  c:\Git\project\bug.py:10:18 - error: Argument of type "str | StandardKey" cannot be assigned to parameter "k1" of type "int" in function "__init__"
    Type "str | StandardKey" is not assignable to type "int"
      "StandardKey" is not assignable to "int" (reportArgumentType)
2 errors, 0 warnings, 0 informations
```

pyright-python v1.1.393, PySide6-Essentials v6.8.2

**Comment by erictraut** [comment]
This is a duplicate of #9784.

**Comment by bersbersbers** [comment]
Interesting - #9784 reports a regression in 1.1.393 after no problem with 1.1.392. By contrast, I can reproduce this issue here going back to 1.1.392, 1.1.390 and even 1.1.360. But I'll subscribe to #9784 and see if that fix applies to my issue as well.

**Comment by erictraut** [comment]
I'm not able to repro the problem with 1.1.392.

**Comment by bersbersbers** [comment]
You are absolutely right! `PYRIGHT_PYTHON_FORCE_VERSION=latest` was to blame :)
